scrapy/utils/url.py

```diff
@@ -110,3 +110,12 @@ def escape_ajax(url):
     if not frag.startswith('!'):
         return url
     return add_or_replace_parameter(defrag, '_escaped_fragment_', frag[1:])
+
+def add_http_if_no_scheme(url):
+    """Adds http as the default scheme if it is missing from the url"""
```

Comment:
A couple minor notes: according to pep8 there should be 2 blank lines before the function; according to pep257 docstring should use imperative mood:

> The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".